### PR TITLE
[MAT-6232] Handle VSAC Comms error during DRC Validation

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## Terminology Service PR

Jira Ticket: [MAT-6232](https://jira.cms.gov/browse/MAT-6232)
(Optional) Related Tickets:

### Summary
Handle Bad Request responses from VSAC that are not validation errors, but normal communication errors when validating a DRC.

VSAC returns a 400 when either the code, code system, or code system version are invalid.

VSAC also returns a 400 for the more common unprocessable request error. This response is formatted differently from its validation counterpart, and was not being recognized as an error.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java [hunk 1]
```diff
@@ -213,15 +213,27 @@ private String buildCodeSystemVersion(CqlCode cqlCode, CodeSystemEntry codeSyste
...
+        cqlCode.getCodeSystem().setValid(false);
+        cqlCode
+            .getCodeSystem()
+            .setErrorMessage(vsacCode.getErrors().getResultSet().get(0).getErrDesc());
+      } else if (errorCode == 802) {
+        cqlCode.setValid(false);
+        cqlCode.setErrorMessage(vsacCode.getErrors().getResultSet().get(0).getErrDesc());
+      }
+    // API Errors appear in the status field.
+    } else if (StringUtils.isNumeric(vsacCode.getStatus())) {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/terminology/service/VsacService.java:230]: so in this scenario, VSAC doesn’t provide an error message?

**Review by adongare — APPROVED** [review]

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java:230]: Technically, it does: ```"error": "Bad Request"```, which is put somewhere different than the validation errors.

Comment (adongare) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java:230]: ah..okay, thanks!
